packages/context/src/lib/controllers/context-provider.ts

```diff
@@ -39,6 +39,17 @@ export interface Options<C extends Context<unknown, unknown>> {
   initialValue?: ContextType<C>;
 }
 
+/**
+ * Generates a public interface type that removes private and protected fields.
+ * This allows accepting otherwise compatible versions of the type (e.g. from
+ * multiple copies of the same package in `node_modules`).
+ */
+type Interface<T> = {
+  [K in keyof T]: T[K];
+};
+
+type ReactiveElementHost = Interface<Omit<ReactiveElement, 'renderRoot'>>;
```

Comment:
Any reason why this can't also just be `ReactiveContorllerHost & HTMLElement` like for the `ContextConsumer`? Or maybe the other way around to use this for `ContextConsumer`?

Also wdyt about upstreaming this type into `reactive-element` so other decorators can use it too?

Reply:
`ReactiveContorllerHost & HTMLElement` works